### PR TITLE
Should we use a negative limit with String.split in substitutions?

### PR DESCRIPTION
**Pull request by solde9**
Calling split with zero or no limit will discard trailing empty fields.
For that reason, s// will not delete previous search. Using a negative
limit allows this to work because it split the definition into [s,"",""]

### CODE REVIEW

**Comment by albertdev** [comment]
Do you have some tests which reproduce the original problem and are fixed with your changes? (You can still add them by pushing an extra commit to the branch in your repo.)

`SearchModeTests` should have the setup you need.

**Comment by solde9** [comment]
I don't know if this is right but I uncommented some code in `net.sourceforge.vrapper.core.tests.utils.TestSearchService.replace()` (https://github.com/vrapper/vrapper/blob/master/net.sourceforge.vrapper.core.tests/src/net/sourceforge/vrapper/core/tests/utils/TestSearchService.java#L53) in an attempt to mock the "replace" functionality of the platform. It doesn't look correct although it works for this particular case. I'm not sure if it's ok to push it. That code allowed me to create this test:

```java
@Test
public void testSubstitution() throws CommandExecutionException {
	content.setText("one two three");
	registerManager.getRegister("/").setContent(new StringRegisterContent(ContentType.TEXT, "two"));
	
	new SubstitutionOperation("s//").execute(adaptor, null, ContentType.LINES);
	assertEquals("one  three", content.getText());
}
```
in `CommandLineTests`.

Basically, the problem happens if you execute `:s//` (find previous search and replace with empty string). The definition is split but the trailing empty fields are discarded (https://github.com/vrapper/vrapper/blob/master/net.sourceforge.vrapper.core/src/net/sourceforge/vrapper/utils/SubstitutionDefinition.java#L23). This doesn't happen if the search string is specified (ex. `:s/word/`) or if a flag is added (ex. `:s//g`) 

Let me know if it's ok to include those test.

**Comment by albertdev** [comment]
Ah, I guess I commented out that code because it doesn't deal with most flags. I will check if I can do anything about it and add the test then.

Thank you for looking into this!

**Comment by albertdev** [comment]
I've merged your commit by hand and wrote some tests as I couldn't get your sample test to work in `CommandLineTests`, there was just too much missing.

I ended up with rewriting the replace method, now we should actually have a test stub which can do real case-(in)sensitive replaces and returns the right number of hits.
